### PR TITLE
Feature/upload only recipe

### LINKED ISSUE #25: build_policy missunderstanding
Hi!

I'm building a package ([here](https://github.com/jgsogo/conan-portaudio)) that download the sources from a git repo. I want to maintain a version, `master`, that always grab the last revision from that repository. I supposed that this will be the behaviour of `build_policy='always'`, the client will get the sources and compile them each time conan install is invoked... but

```
ERROR: Conanfile has build_policy='always', no packages can be uploaded
Traceback (most recent call last):
  File "build.py", line 16, in <module>
    builder.run()
  File "C:\Python27-x64\lib\site-packages\conan\packager.py", line 261, in run
    self._upload_packages()
  File "C:\Python27-x64\lib\site-packages\conan\packager.py", line 367, in _upload_packages
    raise Exception("Error uploading")
Exception: Error uploading
Command exited with code 1
```

It's ok, it has no sense to upload the binaries, but I want to upload the recipe... is this behaviour possible?

Thanks!

### LINKED ISSUE #153: Upload only the recipe
Sometimes you want to test many configurations but upload only the recipe. 
option to upload only the recipe? or a flag in the configurations individually?

### PR DESCRIPTION
**Pull request by uilianries**
Changelog: Feature: Upload only the Conan recipe (#153) (#25)

Add the option **CONAN_UPLOAD_ONLY_RECIPE** which allows to upload only the recipe when is True.

fixes #153 
fixes #25

### CODE REVIEW

File: cpt/packager.py [hunk 1]
```diff
@@ -143,6 +144,8 @@ def __init__(self, username=None, channel=None, runner=None,
         else:
             self.upload_only_when_tag = get_bool_from_env("CONAN_UPLOAD_ONLY_WHEN_TAG")
 
+        self.upload_only_recipe = upload_only_when_tag or get_bool_from_env("CONAN_UPLOAD_ONLY_RECIPE")
```
Comment (madebr) [inline cpt/packager.py]:
```suggestion
        self.upload_only_recipe = upload_only_recipe or get_bool_from_env("CONAN_UPLOAD_ONLY_RECIPE")
```
I don't think passing this argument is tested. The environment variable is tested though.

Comment (uilianries) [reply cpt/packager.py]: Thanks for finding this bug. I'll add a test case to check this.

**Comment by lasote** [comment]
conflicts

**Comment by uilianries** [comment]
@lasote all conflicts have been solved.

File: README.md [hunk 2]
```diff
@@ -798,6 +798,30 @@ You can upload the generated packages automatically to a conan-server using the
         CONAN_STABLE_BRANCH_PATTERN: "release/*"
 
 
+<<<<<<< HEAD
```
Comment (lasote) [inline README.md]: :-1:

Comment (uilianries) [reply README.md]: :man_facepalming:

**Pull request merged**